x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -3237,6 +3239,162 @@ public void testStatsWithCanonicalAggregate() throws Exception {
         assertThat(Expressions.attribute(fields.get(1)), is(Expressions.attribute(sum_argument)));
     }
 
+    /**
+     * Expects
+     * Project[[s{r}#3, s_expr{r}#5, s_null{r}#7, w{r}#10]]
+     * \_Eval[[$$COUNT$s$0{r}#25 * 3[INTEGER] AS s, $$COUNT$s$0{r}#25 * 3.14[DOUBLE] AS s_expr, null[LONG] AS s_null]]
```

Comment:
Ideally, we need to see in a unit test that `sum([1,2])` is in fact `mv_sum * count`.
This test here is ok, but it doesn't really check the transformation we are doing for `sum([1,2])`.

Reply:
Updated - now the tests only check the result of the substitutions, and in particular we check whether we replace `avg` by `mv_avg` etc.